### DISCUSSION TITLE
Connection refused on postgresql with teleport

### DISCUSSION THREAD

**Discussion by flechter** [gravitational/teleport#37172] [Q&A]
we tried to configure teleport to connect to a postgresql database but we have this error : 
root@detcisedt0009:~# tsh db connect example-postgres3
psql: error: connection refused
ERROR: exit status 2
we tryed to follow this tutorial : 
https://www.youtube.com/watch?v=8Hks1JA9JR4&t=2546s

some informations : 
root@detcisedt0009:~# teleport version
Teleport v14.2.0 git:api/v14.2.0-0-g5d4cf8198c go1.21.4

root@detcisedt0009:~# tsh status
> Profile URL:        https://teleportpoc.hosting.cegedim.cloud:443
  Logged in as:       teleport-admin
  Cluster:            teleportpoc.example.com
  Roles:              access, db, editor, postgres-access
  Logins:             adm-sdidierjean
  Kubernetes:         enabled
  Databases:          example-postgres, example-postgres3
  Valid until:        2024-01-24 23:25:20 +0100 CET [valid for 6h50m0s]
  Extensions:         login-ip, permit-agent-forwarding, permit-port-forwarding, permit-pty, private-key-policy

root@detcisedt0009:~# tsh db ls
Name                                Description Allowed Users Labels  Connect                          
----------------------------------- ----------- ------------- ------- -------------------------------- 
> example-postgres3 (user: postg...             [*]           env=dev tsh db connect example-postgres3 

and we have this error on teleport : 
2024-01-24T11:59:06+01:00 ERRO [DB:SERVIC] Failed to handle connection. addr:10.59.12.52:33048 error:[
ERROR REPORT:
Original Error: syscall.Errno connection refused
Stack Trace:
        github.com/gravitational/teleport/lib/srv/db/common/errors.go:159 github.com/gravitational/teleport/lib/srv/db/common.ConvertConnectError
        github.com/gravitational/teleport/lib/srv/db/postgres/engine.go:271 github.com/gravitational/teleport/lib/srv/db/postgres.(*Engine).connect
        github.com/gravitational/teleport/lib/srv/db/postgres/engine.go:143 github.com/gravitational/teleport/lib/srv/db/postgres.(*Engine).HandleConnection
        github.com/gravitational/teleport/lib/srv/db/common/reporting.go:87 github.com/gravitational/teleport/lib/srv/db/common.(*reportingEngine).HandleConnection
        github.com/gravitational/teleport/lib/srv/db/server.go:1021 github.com/gravitational/teleport/lib/srv/db.(*Server).handleConnection
        github.com/gravitational/teleport/lib/srv/db/server.go:945 github.com/gravitational/teleport/lib/srv/db.(*Server).HandleConnection
        github.com/gravitational/teleport/lib/reversetunnel/transport.go:246 github.com/gravitational/teleport/lib/reversetunnel.(*transport).start
        github.com/gravitational/teleport/lib/reversetunnel/agent.go:584 github.com/gravitational/teleport/lib/reversetunnel.(*agent).handleDrainChannels.func2
        runtime/asm_amd64.s:1650 runtime.goexit
User Message: connection refused] db/server.go:947

**Comment by webvictim** [comment]
Can you share the full output of `tsh db ls -v` from the machine you're attempting to connect from?

Also, please share the `db_service` section of your `/etc/teleport.yaml` config file!

**Reply by flechter** [reply]
Hello,
Thanks for the answer, for the command :
root@detcisedt0009:~# tsh db ls -v
Name              Description Protocol Type        URI                                      Allowed Users Labels                                  Connect 
----------------- ----------- -------- ----------- ---------------------------------------- ------------- --------------------------------------- ------- 
example-postgres3             postgres self-hosted detcisedt0009.hosting.cegedim.cloud:5432 [*]           env=dev,teleport.dev/origin=config-file         
and the teleport.yaml : 
version: v3
teleport:
  nodename: detcisedt0009.hosting.cegedim.cloud
  data_dir: /var/lib/teleport
  auth_token: "/tmp/token"
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: "sha256:d551aeeb16276c29f9e1fb9c026abd85b8f9a90d7d8fa06a641621e937b92e75"
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  cluster_name: teleportpoc.example.com
  proxy_listener_mode: multiplex
ssh_service:
  enabled: "yes"
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
db_service:
  enabled: true


  # Lists statically registered databases proxied by this agent.
  databases:
  - name: "example-postgres3"
    protocol: "postgres"
    uri: "detcisedt0009.hosting.cegedim.cloud:5432"
    static_labels:
      "env": "dev"
proxy_service:
  enabled: "yes"
  web_listen_addr: 0.0.0.0:443
  public_addr: teleportpoc.hosting.cegedim.cloud:443
  https_keypairs:
  - key_file: /var/lib/teleport/clepriv2.pem
    cert_file: /var/lib/teleport/chaine2.pem
  https_keypairs_reload_interval: 0s
  acme: {}
